From 0339bdc0a49c415f55f728ac9f069ff426898722 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Wed, 11 Jan 2023 02:31:02 +0530
Subject: [PATCH] Fix implicit argument conversion on UWP

vpx_util\vpx_thread.h(79,13): error C2664: 'HANDLE CreateThread(LPSECURITY_ATTRIBUTES,SIZE_T,LPTHREAD_START_ROUTINE,LPVOID,DWORD,LPDWORD)': cannot convert argument 3 from 'unsigned int (__cdecl *)(void *)' to 'LPTHREAD_START_ROUTINE'
---
 vpx_util/vpx_thread.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/vpx_util/vpx_thread.h b/vpx_util/vpx_thread.h
index 6d308e9..c84e96a 100644
--- a/vpx_util/vpx_thread.h
+++ b/vpx_util/vpx_thread.h
@@ -78,7 +78,8 @@ static INLINE int pthread_create(pthread_t *const thread, const void *attr,
 #ifdef USE_CREATE_THREAD
   *thread = CreateThread(NULL,          /* lpThreadAttributes */
                          0,             /* dwStackSize */
-                         start, arg, 0, /* dwStackSize */
+                         (LPTHREAD_START_ROUTINE) start, arg,
+                         0,             /* dwStackSize */
                          NULL);         /* lpThreadId */
 #else
   *thread = (pthread_t)_beginthreadex(NULL,          /* void *security */
-- 
2.37.1 (Apple Git-137.1)

